src/components/vm/nics/vmNicsCard.jsx

```diff
@@ -392,28 +448,51 @@ export class VmNetworkTab extends React.Component {
                         actionName: _("Remove"),
                         deleteHandler: () => domainDetachIface({ connectionName: vm.connectionName, index: network.index, vmName: vm.name, live: vm.state === 'running', persistent: vm.persistent && nicPersistent }),
                     };
-                    const deleteNICAction = (
+                    const disabled = vm.state != 'shut off' && vm.state != 'running';
+
+                    let button = (
                         <DeleteResourceButton objectId={`${id}-iface-${networkId}`}
-                                              disabled={vm.state != 'shut off' && vm.state != 'running'}
+                                              disabled={disabled}
                                               dialogProps={deleteDialogProps}
                                               actionName={_("Remove")}
                                               overlayText={_("The VM needs to be running or shut off to detach this device")}
-                                              isSecondary />
+                                              isDropdownItem />
                     );
 
+                    if (disabled) {
+                        button = (
+                            <Tooltip id={`delete-${id}-tooltip`}
+                                     key={`delete-${id}-tooltip`}
+                                     content={_("The VM needs to be running or shut off to detach this device")}>
+                                <span>{button}</span>
+                            </Tooltip>
+                        );
+                    }
+
+                    const dropdownItems = [
+                        button,
+                    ];
+
                     return (
                         <div className='machines-listing-actions'>
-                            {deleteNICAction}
                             <Button id={`${id}-iface-${networkId}-` + (isUp ? 'unplug' : 'plug')}
                                     variant='secondary'
                                     onClick={onChangeState(network)}>
                                 {isUp ? 'Unplug' : 'Plug'}
                             </Button>
                             {editNICAction()}
+                            <Dropdown onSelect={() => this.setState({ isDropdownOpen: false })}
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1077-20230626-085745-226f7c4b-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)